src/node/util.ts

```diff
@@ -241,8 +249,71 @@ export function getSystemErrorMessage(): void {
   throw new Error('node:util getSystemErrorMessage is not implemented');
 }
 
-export function styleText(): void {
-  throw new Error('node:util styleText is not implemented');
+function escapeStyleCode(code: number | undefined): string {
+  if (code === undefined) return '';
+  return `\u001b[${code}m`;
+}
+
+// TODO(gbedford): We do not yet implement process.stdout, so to ensure the correct
+// behaviour, we use a placeholder value internally.
+const stdoutPlaceholder = Object.create(null);
+export function styleText(
+  format: string | string[],
+  text: string,
+  {
+    validateStream = true,
+    stream = stdoutPlaceholder,
+  }: { validateStream?: boolean; stream?: ReadableStream | WritableStream } = {}
+): string {
+  validateString(text, 'text');
+  validateBoolean(validateStream, 'options.validateStream');
+
+  let skipColorize;
+  if (validateStream) {
+    if (
+      !isReadableStream(stream) &&
+      !isWritableStream(stream) &&
+      !isNodeStream(stream) &&
+      stream !== stdoutPlaceholder
+    ) {
+      throw new ERR_INVALID_ARG_TYPE(
+        'stream',
+        ['ReadableStream', 'WritableStream', 'Stream'],
+        stream
+      );
+    }
+
+    // If the stream is falsy or should not be colorized, set skipColorize to true
+    skipColorize =
+      (stream as any)?.isTTY &&
```

Comment:
We tend to avoid using any as much as we can. For example, you expect stream to be an object in here, but if you pass `true` it will throw an error